packages/parser/src/fs.ts

```diff
@@ -34,6 +34,27 @@ export async function load(filepath: string, themeMeta?: SlidevThemeMeta, conten
     const baseSlide = data.slides[iSlide]
     if (!baseSlide.frontmatter.src) {
       iSlide++
+
+      baseSlide.content = baseSlide.content.replaceAll(
+        /^```(\w+?)\s*\[([\s\S]+?)\]([\s\S]*?)\n+^```/mg,
+        (full, lang = '', external: string, rest: string) => {
+          const [externalPath, externalRangeStr] = external.split(':').map(i => i.trim())
+          const sourcePath = resolve(dir, 'snippets', externalPath)
+          entries.add(sourcePath)
+          let source: string
+          if (!existsSync(sourcePath)) {
+            lang = 'plaintext'
+            source = `File not found: ${sourcePath}`
+          }
+          source = readFileSync(sourcePath, 'utf-8')
```

Comment:
It may throw some other errors here. Catch them here?

Reply:
Resolved in 30e77d5